MaterialLib/Fluid/Viscosity/PressureDependentViscosity.h

```diff
@@ -0,0 +1,64 @@
+/*!
+   \file  PressureDependentViscosity.h
+   \brief Declaration of class for the pressure dependent viscosity model.
+
+   \author Wenqing Wang
+   \date Jan 2015
+
+   \copyright
+    Copyright (c) 2012-2015, OpenGeoSys Community (http://www.opengeosys.org)
+               Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+*/
+#ifndef PRESSURE_DEPENDENT_VISCOSITY_H_
+#define PRESSURE_DEPENDENT_VISCOSITY_H_
+
+#include <string>
+#include <algorithm>    // std::max
+
+#include "ViscosityType.h"
+
+namespace MaterialLib
+{
+
+/// Pressure dependent viscosity model,
+class PressureDependentViscosity
```

Comment:
I like the idea in general. In ogs5 we have got several pressure dependent viscosity models numbered from 0 to 4 or something. It would be nice if we could avoid five classes having different ids.
Is there a generalization to this?

There are also models which are dependent on p only, on T only, on p and T, on p and T and t etc. So how to avoid a possible model name "PressureTemperatureTimeDependentViscosity7a"?